src/module/Provider.sol

```diff
@@ -72,6 +75,19 @@ contract Provider is IProvider {
             return IsfrxETH(MC.SFRXETH).pricePerShare() * frxETHPriceInETH / 1e18;
         }
 
+        if (asset == MC.CURVE_LP_YNETH_YNLSDE_STRATEGY) {
+            (int256 lpRate, uint256 updatedAt) = ICurveLpConnector(MC.CURVE_LP_YNETH_YNLSDE_CONNECTOR).rate();
+            if (lpRate < 0) {
+                revert RateIsNegative();
+            }
+            // solhint-disable-next-line not-rely-on-time
+            if (updatedAt < block.timestamp - 24 hours) {
+                revert RateIsStale();
+            }
+
+            return uint256(lpRate);
```

Comment:
the rate required here is not the rate of the LP but the rate of the strategy

so far you have the rate of the LP, you need to combine the 2 rates, the rate of the strategy with the rate of the lp

Reply:
@danoctavian I know we discussed that. But this is actually already calculated 

https://github.com/yieldnest/curve-lp-ynethx/blob/04ff958bc371375c8da74d771723024578d05aec/src/test/Connector.t.sol#L32

If you see here the `YvCurveLpOracle` oracle is used in the deployment of Connector.sol 

https://github.com/yieldnest/curve-lp-ynethx/blob/04ff958bc371375c8da74d771723024578d05aec/src/periphery/YvCurveLpOracle.sol

And the convertToShares is already done in this oracle.

Reply:
I renamed the variable to `strategyRate` for clarity.